includes/config.inc

```diff
@@ -454,6 +454,7 @@ function islandora_solr_metadata_management($data) {
         ),
         'weight' => array(
           '#type' => 'weight',
+          '#delta' => $delta >= 10 ? $delta : 10,
```

Comment:
I'm... not sure efficiency should really be an issue, here...

Really, just calculate the max delta once, and have it be used in every iteration... Using `max()` would be more obvious (and have an arbitrary number fewer times). Something like:

```php
$max_delta = max(count($data), 10);
$weight = 0;
$map = function ($field) use ($max_delta, &$weight) {
  [...]
  'weight' => array(
    '#type' => 'weight',
    '#delta' => $max_delta,
  [...]
};
```